src/main/java/com/jfrog/ide/idea/scan/ScanManagersFactory.java

```diff
@@ -6,28 +6,38 @@
 import com.intellij.openapi.project.DumbService;
 import com.intellij.openapi.project.Project;
 import com.intellij.openapi.projectRoots.Sdk;
+import com.jfrog.ide.common.persistency.XrayScanCache;
+import com.jfrog.ide.common.scan.BulkScanLogic;
+import com.jfrog.ide.common.scan.EmptyScanLogic;
+import com.jfrog.ide.common.scan.GraphScanLogic;
+import com.jfrog.ide.common.scan.ScanLogic;
 import com.jfrog.ide.common.utils.PackageFileFinder;
 import com.jfrog.ide.idea.configuration.GlobalSettings;
 import com.jfrog.ide.idea.log.Logger;
 import com.jfrog.ide.idea.navigation.NavigationService;
 import com.jfrog.ide.idea.ui.ComponentsTree;
 import com.jfrog.ide.idea.ui.LocalComponentsTree;
 import com.jfrog.ide.idea.utils.Utils;
+import com.jfrog.xray.client.impl.XrayClient;
 import org.jetbrains.annotations.NotNull;
 
 import java.io.IOException;
+import java.nio.file.Files;
 import java.nio.file.Path;
+import java.nio.file.Paths;
 import java.util.Collection;
 import java.util.Map;
 import java.util.Set;
 
 import static com.jfrog.ide.common.log.Utils.logError;
+import static com.jfrog.ide.common.utils.XrayConnectionUtils.createXrayClientBuilder;
 
 /**
  * Created by yahavi
  */
 public class ScanManagersFactory {
 
+    private static final Path HOME_PATH = Paths.get(System.getProperty("user.home"), ".jfrog-idea-plugin");
```

Comment:
The above variable is retrieved multiple times in the code, so the following is duplicated - 
```
Paths.get(System.getProperty("user.home"), ".jfrog-idea-plugin");
```
Let's create a utility method or methods (in Utils), to retrieve all the needed directories under the user home.